macros/sql_gen/gen_final_cte.sql

```diff
@@ -17,10 +17,30 @@
 
         select * from final 
 
+            -- metric where clauses...
+        {% if where %}
+        where (
+            {%- for filter in where %}
+                {{ filter }}
+                {% if not loop.last %} and {% endif %}
+            {%- endfor %}
+        )
+        {% endif %}
+
     {% else %}
 
     select * from joined_metrics
 
+    -- metric where clauses...
+    {% if where %}
+    where (
+        {%- for filter in where %}
+            {{ filter }}
+            {% if not loop.last %} and {% endif %}
+        {%- endfor %}
+    )
+    {% endif %}
+
```

Comment:
I wrote the below comment first, but I just expanded more of the diff and realised that this isn't an adapter-specific implementation, it's actually just the same code catering for four different outcomes of if statements. So I double down on the "how could this be tidier" challenge (happy to pair on you with it!)